truffle/src/main/ruby/core/process_mirror.rb

```diff
@@ -109,6 +110,7 @@ def initialize(env_or_cmd, *args)
 
             @command = command
             @argv = argv
+            @env_array = []
```

Comment:
I would not initialize it, so it remains nil until it's properly set and not empty.